app/models/book.rb

```diff
@@ -12,12 +13,22 @@ def self.order_by(field, list_order)
     end
   end
 
-  def self.sortable_fields
+  def self.sort_options
+    options = Book.searchable_fields.map { |field| [field.to_s.humanize, field.to_s] }
+    options << ["Most popular"]
+    options
+  end
+
+  def self.searchable_fields
     [:title, :published_date, :author, :price, :category]
   end
 
   def self.search(query)
-    search_clauses = sortable_fields.map { |field| "#{field} LIKE :q" }
+    search_clauses = searchable_fields.map { |field| "#{field} LIKE :q" }
     where(search_clauses.join(' OR '), q: "%#{query}%")
   end
+
+  def times_sold
+    order_items.sum(:quantity)
```

Comment:
This works, but I think there's a better way to do it. Check out [counter caches](http://yerb.net/blog/2014/03/13/three-easy-steps-to-using-counter-caches-in-rails/). While it does de-normalize your database, it's more efficient.